sticht/rollbacks/sources/splunk.py

```diff
@@ -116,52 +119,46 @@ def process_result(self, result: Optional[List[Dict[Any, Any]]], earliest_timest
         number of results from a query against configured thresholds to determine
         whether or not to rollback or not
         """
-        if earliest_timestamp_seconds != 'now':
-            if earliest_timestamp_seconds < self.start_timestamp_seconds:
-                self.bad_before_mark = self.is_window_bad(result)
-        else:
+        if earliest_timestamp_seconds != 'now' and earliest_timestamp_seconds > self.start_timestamp_seconds:
             self.bad_after_mark = self.is_window_bad(result)
+        else:
+            self.bad_before_mark = self.is_window_bad(result)
 
         old_failing = self.failing
+
         self.failing = self.bad_after_mark and not self.bad_before_mark
+
         if self.failing == (not old_failing):
             self.on_failure_callback(self)
 
     def is_window_bad(self, result) -> bool:
         if self._query_type == 'results':
-            if self._lower_bound and len(result) < self._lower_bound:
-                return True
-            elif self._upper_bound and len(result) > self._upper_bound:
-                return True
-        else:
-            result_num = next(iter(result[0].values()))
-            if self._lower_bound and result_num < self._lower_bound:
-                return True
-            elif self._upper_bound and result_num > self._upper_bound:
-                return True
-        return False
```

Comment:
Sorry I'm slightly far removed from this code, but what does it mean if `self._query_type != 'results'` here, and why would that mean we should always `return True` now compared to the previous logic, where we only returned true if the first result was out of bounds?

Reply:
There are two types, and I completely just missed the `number` type. It should not always `return True`. Must have removed it somehow. Good catch!